ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java

```diff
@@ -158,69 +188,85 @@ public void getTableDetail(Context ctx) {
+
+    // getRuntime table from catalog
+    MixedHiveCatalogImpl arcticHiveCatalog = (MixedHiveCatalogImpl) tableService.getServerCatalog(catalog);
+
+    TableIdentifier tableIdentifier = TableIdentifier.of(catalog, db, table);
+    HiveTableInfo hiveTableInfo;
+    try {
+      Table hiveTable = HiveTableUtil.loadHmsTable(arcticHiveCatalog.getHiveClient(), tableIdentifier);
+      List<AMSColumnInfo> schema = transformHiveSchemaToAMSColumnInfo(hiveTable.getSd().getCols());
+      List<AMSColumnInfo> partitionColumnInfos = transformHiveSchemaToAMSColumnInfo(hiveTable.getPartitionKeys());
+      hiveTableInfo = new HiveTableInfo(tableIdentifier, TableMeta.TableType.HIVE, schema, partitionColumnInfos,
+          new HashMap<>(), hiveTable.getCreateTime());
+    } catch (Exception e) {
+      LOG.error("Failed to getRuntime hive table info", e);
+      ctx.json(new ErrorResponse(HttpCode.BAD_REQUEST, "Failed to getRuntime hive table info", ""));
+      return;
+    }
+    ctx.json(OkResponse.of(hiveTableInfo));
     ctx.json(OkResponse.ok());
   }
 
   /**
    * upgrade hive table to arctic.
    */
   public void upgradeHiveTable(Context ctx) {
-    // String catalog = ctx.pathParam("catalog");
-    // String db = ctx.pathParam("db");
-    // String table = ctx.pathParam("table");
-    // UpgradeHiveMeta upgradeHiveMeta = ctx.bodyAsClass(UpgradeHiveMeta.class);
-    //
-    // ServerCatalog serverCatalog = CatalogBuilder.buildServerCatalog(tableService.getCatalogMeta(catalog));
-    //
-    // String thriftHost = serviceConfig.getString(ArcticManagementConf.THRIFT_BIND_HOST);
-    // Integer thriftPort = serviceConfig.getInteger(ArcticManagementConf.THRIFT_BIND_PORT);
-    // ArcticHiveCatalog arcticHiveCatalog
-    //     = (ArcticHiveCatalog) CatalogUtil.getArcticCatalog(thriftHost, thriftPort, catalog);
-    // try {
-    //   UpgradeHiveTableUtil.upgradeHiveTable(arcticHiveCatalog, TableIdentifier.of(catalog, db, table),
-    //       upgradeHiveMeta.getPkList()
-    //           .stream()
-    //           .map(UpgradeHiveMeta.PrimaryKeyField::getFieldName)
-    //           .collect(Collectors.toList()), upgradeHiveMeta.getProperties());
-    //   ctx.json(OkResponse.ok());
-    // } catch (Exception e) {
-    //   LOG.error("upgrade hive table error:", e);
-    //   ctx.json(new ErrorResponse(HttpCode.BAD_REQUEST, "Failed to upgrade hive table", ""));
-    // }
+    String catalog = ctx.pathParam("catalog");
+    String db = ctx.pathParam("db");
+    String table = ctx.pathParam("table");
+    Preconditions.checkArgument(
+        !catalog.isEmpty() && !db.isEmpty() && !table.isEmpty(),
+        "catalog.database.tableName can not be empty in any element");
+    UpgradeHiveMeta upgradeHiveMeta = ctx.bodyAsClass(UpgradeHiveMeta.class);
+
+    ArcticHiveCatalog arcticHiveCatalog
+        = (ArcticHiveCatalog) CatalogLoader.load(String.join("/", AmsUtil.getAMSThriftAddress(serviceConfig), catalog));
+    try {
+      new Thread(() -> {
```

Comment:
Will changing the upgrade process to asynchronous execution affect the user experience? For example, users may not be able to perceive the upgrade results correctly.